### ISSUE TITLE
Support Video Layer 

### ISSUE THREAD

**Issue by easylogic** [easylogic/editor#54]
* [x] add video layer 
* [x] support video asset 
* [x] play, volume control 
* [x] support to play video in timline

**Comment by j2l** [comment]
Great to see that you're pushing hard to make it!
Thank you

**Comment by easylogic** [comment]
I released v0.9.50

**Comment by easylogic** [comment]
Minimal control function added. Only the starting point can be specified in the timeline. Please tell me if there is a function you need.

**Comment by j2l** [comment]
I'll try the video as soon as I can this week.
Thank you.

**Comment by j2l** [comment]
OK,  I squeezed in time for this great project.
Video is playing on Chrome and FF (webm, mp4). Matroska is not supported (mkv, would be useful, not sure if it's a browser problem).

The idea of adding the play time distance start and end in the timeline is wonderful!

Though, when linking it to the timeline, I noticed that the timeline play button is not related to the video play control: if you click pause in the timeline, it doesn't pause, it continues to play, beyond the limit in the timeline. Best would be to be able to move the play head in the timeline (and video plays and pause at the right time) to find cue points for svg.
Since the goal is to draw above video to animate SVG in sync (rotoscoping to get natural animations), it won't work as is.

More globally, as soon as you click on the timeline, it looses the focus of the selected layer. It's just a glitch, when you know it, you re-click on the layer. But it's confusing at first.

Hope it helps!

PS: I tried Sapa and really don't understand how it produces a result.
Would you please show me a quick example?

**Comment by easylogic** [comment]
Let's create a sample that controls the video in the timeline. 
It would be nice to see it and discuss it together.

**Comment by easylogic** [comment]
I released v0.9.51 .  

If you stop the timeline while the video is playing in the timeline, the video also stops. If you press Resume, playback starts from that point.

**Comment by j2l** [comment]
Hello Jinho (I hope I can call you Jinho, please let me know. I'm Phil BTW),

Now, the pause/play button works great!

Thank you very much for your work. being able to draw and animate SVG upon video is very useful and hardly seen in a web app.

About the video, and only if you want, I think it may need a little more work (timeline play head scrubbing working like style > video > play head scrubbing, fix timeline head scrubbing playing the video forward) but it's already very good enough to draw!

Oh, I notice that I don't know how to animate SVG points by themselves, for example:
![draw_on_video](https://user-images.githubusercontent.com/65325/84496832-efc1da00-acad-11ea-9899-72146c8afc8a.png)
How to animate only this (blue, selected) point, or a selection of points?

**Comment by easylogic** [comment]
hi Phil  ( My name is Jinho. You can call me Jinho. )

Path can be animated on timeline. 

This can be a little complicated, so I'll show you a video

I'll let you know when it's organized.

**Comment by easylogic** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/591983/84511904-304b4300-ad02-11ea-81fa-d70c35ecedf9.gif)


Basically, you can drive the path animation in the timeline. 

However, if you move the points, the entire boundary changes, so the path is controlled in an unexpected way. 

This seems to have to be fixed.

**Comment by j2l** [comment]
Thank you for your recording!
I don't really get how you record your changes on the timeline, the gif stops before it's t"weened" and played.
When I try to record the first point positions on the timeline, then move a point, I can double click on the timeline, but it doesn't seem to record the new position and the curve is in intermediate state, waiting for an action I guess:
![point_timeline](https://user-images.githubusercontent.com/65325/84572836-098e1a80-ad9d-11ea-8d1c-4932f02a25a9.png)
Your helpers, handles and guides are wonderful. 
I really hope I can animate a simple path points soon :)

**Comment by j2l** [comment]
I'm not sure I understand what M, C, Z, Q stand for in the values.
It seems, M is starting point. C seems to be handle position (absolute X,Y, not related to its point) 
Q is for handles next to a point without handles (one handle section). Z is for open or closed.
Is it right?
What does M, C, Q, Z mean?

If point animation is not implemented yet (I'm not sure if I didn't get it or it's missing); may be adding the "diamond" near the point number near X,Y positions (as in red below) to add "diamonded" points to the timeline?
Adding the point number would greatly help as in too (green below). And if M is arbitrary, maybe change it to 1?
![values](https://user-images.githubusercontent.com/65325/84573469-e6656a00-ada0-11ea-85a3-374f602a7c2d.png)

**Comment by easylogic** [comment]
Sorry. I forgot the second video and didn't upload it.

Once you have only the path area, you can create animation.

You can ignore this because the value on the right side of the timeline is for a more detailed area.

Basically, only path editing can be used to divide the previous and subsequent settings.
![path-animation](https://user-images.githubusercontent.com/591983/84591209-8ab9e000-ae77-11ea-927b-32a9a949bd7d.gif)

path is treated as a single attribute value called d.

In other words, you can set values ​​just like any other property.

After selecting a keyframe, you can update the data of the changed path through the refresh button.

**Comment by easylogic** [comment]
However, there is a slightly ambiguous feature here.

When you edit points on a path, the value of the boundary surrounding the path changes.

In other words, the total values ​​of x and y are changed. So, the value of the first set path starts from the wrong x, y.

I think we should reconsider this part again.

**Comment by easylogic** [comment]
I released v0.9.52 

The drawing area is fixed when editing the path. Since the coordinates of the path are modified within that area, the entire path will not move elsewhere when you insert the animation.

**Comment by j2l** [comment]
Great; thank you!
I used Firefox and Chromium on Ubuntu.
